Remove client side navigation from Products beta block

imanish003 opened the pull request:
Changes:
- Removed the `add_navigation_id_directive` method and its associated filter. This method previously added a `data-wc-navigation-id` attribute to the query block for client-side navigation, which is no longer required.
- Removed the `add_navigation_link_directives` method and its associated filter. This method previously added interactive directives to pagination links inside the Query Pagination block. This specific functionality has been removed.
- Introduced a new method `mark_block_as_interactive`, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.


#### Other Checks

- [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Create a new post
2. Add Products (Beta) block & publish it
3. On Frontend, verify that
    1. Client-side navigation isn't there anymore i.e. changing the page should also refresh the page
    2. Verify that Product button(Add to cart) still works as expected.

Here is a quick demo:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/884b3e32-cabd-425e-b993-ca985dc3308b

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [X] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental


### Changelog

> Remove client side navigation from Products beta block

imanish003 committed:
Remove client side navigation from Products beta block

Changes:
- Removed the `add_navigation_id_directive` method and its associated filter. This method previously added a `data-wc-navigation-id` attribute to the query block for client-side navigation, which is no longer required.
- Removed the `add_navigation_link_directives` method and its associated filter. This method previously added interactive directives to pagination links inside the Query Pagination block. This specific functionality has been removed as it is no longer needed.
- Introduced a new method `mark_block_as_interactive`, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.

gigitux reviewed (approved):
LGTM

luisherranz commented:
> Introduced a new method mark_block_as_interactive, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.

Instead of doing this, if the Products (beta) block doesn't have any interactivity in itself, you should add `data-wc-interactive` to the Product Button blocks instead. That way, the Product Button blocks will also work when they are used outside of the Products (beta) block (like in the product template, if that is a possibility).

imanish003 commented:
> > Introduced a new method mark_block_as_interactive, which is designed to mark the Product Query as an interactive region so that interactive elements can work inside it. This is achieved using the 'data-wc-interactive' attribute.
> 
> Instead of doing this, if the Products (beta) block doesn't have any interactivity in itself, you should add `data-wc-interactive` to the Product Button blocks instead. That way, the Product Button blocks will also work when they are used outside of the Products (beta) block (like in the product template, if that is a possibility).

I think it’s much better approach to have `data-wc-interactive` on all interactive elements, in our case Product Button. Let me try doing it. Thanks 🙌🏻

gigitux committed:
add data-wc-interactive to product button block

imanish003 committed:
Remove unnecessary code

As we have added `data-wc-interactive` on Product button therefore we don't need to add it in Product Query block.

imanish003 committed:
Merge branch 'trunk' into fix/remove-client-side-navigation-from-products-block

luisherranz commented on src/BlockTypes/ProductButton.php:
```diff
@@ -192,7 +192,7 @@ protected function render( $attributes, $content, $block ) {
 				'woocommerce_loop_add_to_cart_link',
 				strtr(
-					'<div class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
+					'<div data-wc-interactive=true class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
```
It's a boolean attribute, you don't need it to be true.

```suggestion
					'<div data-wc-interactive class="wp-block-button wc-block-components-product-button {classes} {custom_classes}"
```

imanish003 replied:
Good catch! Made the changes in 088604d, thanks 🙌🏻

luisherranz reviewed (approved):
Tested and it works fine 🙂👍️

imanish003 committed:
Boolean attribute doesn't need explicit value equal to true

nielslange committed:
Skip failing i18n-related E2E tests

nielslange committed:
Skip failing i18n-related E2E test
